ConfigSource fails to load while executing maven verify

### Describe the bug

When i execute the TEST in eclipse it;s working as expected. But when i execute the request
in maven context it fails.

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.557 s <<< FAILURE! - in com.example.ExampleIT
[ERROR] com.example.ExampleIT.shouldWhen  Time elapsed: 0.008 s  <<< ERROR!
java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSource: com.example.InMemoryConfigSource not a subtype
Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSource: com.example.InMemoryConfigSource not a subtype

### Expected behavior

Should load the ConfigSource.

### Actual behavior

[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.example.ExampleIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.218 s <<< FAILURE! - in com.example.ExampleIT
[ERROR] com.example.ExampleIT.shouldWhen  Time elapsed: 0.006 s  <<< ERROR!
java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSource: com.example.InMemoryConfigSource not a subtype
Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSource: com.example.InMemoryConfigSource not a subtype

### How to Reproduce?

[example.zip](https://github.com/quarkusio/quarkus/files/7839381/example.zip)

1a  Runt he exampleIT test in eclipse (works)

1b Run maven verify (fails)


### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19043.1415]

### Output of `java -version`

openjdk version "11.0.12" 2021-07-20 LTS OpenJDK Runtime Environment Corretto-11.0.12.7.1 (build 11.0.12+7-LTS) OpenJDK 64-Bit Server VM Corretto-11.0.12.7.1 (build 11.0.12+7-LTS, mixed mode

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3

### Additional information

_No response_

--- a/docs/src/main/asciidoc/building-my-first-extension.adoc
+++ b/docs/src/main/asciidoc/building-my-first-extension.adoc
@@ -172,7 +172,8 @@ Your extension is a multi-module project. So let's start by checking out the par
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <quarkus.version>{quarkus-version}</quarkus.version>
-    <surefire-plugin.version>2.22.2</surefire-plugin.version>
+    <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
+    <failsafe.useModulePath>false</failsafe.useModulePath>
   </properties>
   <dependencyManagement>
     <dependencies>

--- a/docs/src/main/asciidoc/jreleaser.adoc
+++ b/docs/src/main/asciidoc/jreleaser.adoc
@@ -631,6 +631,7 @@ As a reference, these are the full contents of the `pom.xml`:
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
     <quarkus.platform.version>{quarkus-version}</quarkus.platform.version>
     <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
+    <failsafe.useModulePath>false</failsafe.useModulePath>
   </properties>
   <dependencyManagement>
     <dependencies>

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/pom.tpl.qute.xml
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/extension-base/java/pom.tpl.qute.xml
@@ -38,6 +38,7 @@
         {#if maven.surefire-plugin.version}
         <surefire-plugin.version>{maven.surefire-plugin.version}</surefire-plugin.version>
         <failsafe-plugin.version>$\{surefire-plugin.version}</failsafe-plugin.version>
+        <failsafe.useModulePath>false</failsafe.useModulePath>
         {/if}
     </properties>
 

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/maven/base/pom.tpl.qute.xml
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/buildtool/maven/base/pom.tpl.qute.xml
@@ -18,6 +18,7 @@
         {/if}
         <compiler-plugin.version>{maven-compiler-plugin.version}</compiler-plugin.version>
         <surefire-plugin.version>{maven-surefire-plugin.version}</surefire-plugin.version>
+        <failsafe.useModulePath>false</failsafe.useModulePath>
         {#if uberjar}
         <quarkus.package.type>uber-jar</quarkus.package.type>
         {/if}

